### PR TITLE
Bugfix change field type work hours to duration field

### LINKED ISSUE #213: Change the field type of `work_hours` to `DurationField`
Original cause explained in https://github.com/Code-Poets/sheetstorm/pull/137/files#r285399800

Change the field type of `work_hours` to `DurationField` https://docs.djangoproject.com/en/2.2/ref/models/fields/#durationfield:
1. Create new field of type `DurationField` with different name.
2. Prepare data migration to migrate from `work_hours` to the new field.
3. Remove `work_hours` field.
4. Rename new field to `work_hours`.
5. Make sure all tests passed.
6. It would be great to make some manual testings before merging.

Yes, it would require 4 migrations to do this correctly. It would be also great to review current data and make sure that everyone using the system meant the same when adding hours, so we know how to migrate data correctly.

### PR DESCRIPTION
**Pull request by kbeker**
Resolves #213

### CODE REVIEW

File: employees/templatetags/data_display_filters.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 from django import template
 
 register = template.Library()
 
 
 @register.filter
-def decimal_to_hours(data: Decimal) -> str:
-    return str(data).replace(".", ":")
+def duration_field_to_string(data: timedelta) -> str:
+    return "{:02d}:{:02d}".format((data.seconds // 3600), (data.seconds % 3600) // 60)
```
Comment (dybi) [inline employees/templatetags/data_display_filters.py]: duplicates:
```python
 @property
    def work_hours_str(self) -> str:
        return self.work_hours.to_eng_string().replace(".", ":")
        return "{:02d}:{:02d}".format((self.work_hours.seconds // 3600), (self.work_hours.seconds % 3600) // 60)
```

File: employees/tests/test_unit_report_model.py [hunk 2]
```diff
@@ -128,74 +125,60 @@ def test_report_model_work_hours_field_should_not_be_empty(self):
...
-
     # PARAM
     def test_report_model_work_hours_field_should_not_accept_value_exceeding_set_minimum(self):
-        self.field_should_not_accept_input("work_hours", ReportModelConstants.MIN_WORK_HOURS.value - Decimal("0.01"))
-
-    # PARAM
-    def test_report_model_work_hours_field_should_not_accept_decimal_value_exceeding_set_maximum(self):
         self.field_should_not_accept_input(
-            "work_hours", ReportModelConstants.MAX_WORK_HOURS_DECIMAL_VALUE.value + Decimal("0.01")
+            "work_hours", ReportModelConstants.MIN_WORK_HOURS.value - datetime.timedelta(seconds=59)
```
Comment (dybi) [inline employees/tests/test_unit_report_model.py]: maybe 
```suggestion
            "work_hours", ReportModelConstants.MIN_WORK_HOURS.value - datetime.timedelta(seconds=1)
```
?

File: employees/common/strings.py [hunk 3]
```diff
@@ -105,3 +101,4 @@ class ReportValidationStrings(NotCallableMixin, Enum):
     WORK_HOURS_SUM_FOR_GIVEN_DATE_FOR_SINGLE_AUTHOR_EXCEEDED = _(
         "Sum of work hours of all reports from a given day for single author must not exceed 24."
     )
+    WORK_HOURS_MIN_VALUE_NOT_EXCEEDED = _("Minimum value for work hours for single report must be bigger than 1 minute")
```
[the remark this answers is not in the document]
Comment (kbeker) [reply employees/common/strings.py]: Right, done

File: employees/models.py [hunk 4]
```diff
@@ -101,11 +88,20 @@ def markdown_description(self) -> str:
     def clean(self) -> None:
         super().clean()
 
-        if (
-            hasattr(self, "author")
-            and isinstance(self.work_hours, Decimal)
-            and self.author.report_set.get_report_work_hours_sum_for_date(self.date, self.pk) + self.work_hours > 24
-        ):
-            raise ValidationError(
+        if hasattr(self, "author") and isinstance(self.work_hours, timedelta):
```
[the remark this answers is not in the document]
Comment (kbeker) [reply employees/models.py]: Done

[the remark this answers is not in the document]
Comment (kbeker) [reply employees/models.py]: Actually, it has to be, but it substituted with validation error if instance does not match

File: employees/templatetags/data_display_filters.py (hunk 1 above)
Comment (kbeker) [reply employees/templatetags/data_display_filters.py]: Moved to common file

File: employees/tests/test_unit_report_model.py (hunk 2 above)
Comment (kbeker) [reply employees/tests/test_unit_report_model.py]: Sure, why not :) Done

**Pull request merged**